test/cpp_extensions/open_registration_extension/pytorch_openreg/csrc/Module.cpp

```diff
@@ -0,0 +1,17 @@
+#include <OpenReg.h>
+
+// Make this a proper CPython module
+static struct PyModuleDef openreg_C_module = {
+    PyModuleDef_HEAD_INIT,
+    .m_name = "pytorch_openreg._C",
+};
+
+PyMODINIT_FUNC PyInit__C(void) {
+    PyObject* mod = PyModule_Create(&openreg_C_module);
+
+    py::object openreg_mod = py::module_::import("pytorch_openreg");
+    // Intentionally leaked
+    openreg::set_impl_registry(openreg_mod.attr("_IMPL_REGISTRY").ptr());
```

Comment:
Is it though? It sure doesn't look like you're taking an owning reference to this

Reply:
Ho yes my bad. Moving this to be borrowed for now.